### PR TITLE
Di 22963 new android app

### PR DESCRIPTION
**Pull request by nadavPepperi**

### CODE REVIEW

File: PepperiZebraPrinter/app/src/main/java/com/pepperi/printer/view/fragments/Main/MainFragment.kt [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        binding.fab.setOnClickListener {
+            findNavController().navigate(R.id.action_MainFragment_to_SecondFragment)
+        }
+
+        setPrinter()
```
Comment (Chasky-H) [inline PepperiZebraPrinter/app/src/main/java/com/pepperi/printer/view/fragments/Main/MainFragment.kt:84]: setPrinters

File: PepperiZebraPrinter/app/src/main/java/com/pepperi/printer/model/Repository.kt [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    private fun hardCodedList(): List<UserPrinter> {
+
+        val newList = arrayListOf <UserPrinter>()
+
+//        newList.add(UserPrinter(0,"test1"))
+//        newList.add(UserPrinter(1,"test2"))
+
+        return newList
+    }
+
```
Comment (Chasky-H) [inline PepperiZebraPrinter/app/src/main/java/com/pepperi/printer/model/Repository.kt:20]: add observeFunction so that viewModels can observe changes

File: PepperiZebraPrinter/app/src/main/java/com/pepperi/printer/view/adapters/ListPrinterAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+
+import com.pepperi.printer.databinding.CardPrinterBinding
+import com.pepperi.printer.model.entities.UserPrinter
+
+class ListPrinterAdapter :
+    ListAdapter<UserPrinter,ListPrinterAdapter.ListViewHolder>(ListMessageDiffCallback()){
+    private var list: List<UserPrinter> = mutableListOf()
```
Comment (Chasky-H) [inline PepperiZebraPrinter/app/src/main/java/com/pepperi/printer/view/adapters/ListPrinterAdapter.kt:15]: why is this needed?

File: PepperiZebraPrinter/app/src/main/java/com/pepperi/printer/view/adapters/ListPrinterAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    override fun onBindViewHolder(holder: ListViewHolder, position: Int) {
+        holder.bind(list[position], position)
+    }
+
+    override fun submitList(submitList: List<UserPrinter>?) {
+        submitList?.let {
+            list = it
+        }
+        super.submitList(list)
+    }
```
Comment (Chasky-H) [inline PepperiZebraPrinter/app/src/main/java/com/pepperi/printer/view/adapters/ListPrinterAdapter.kt:37]: consider removing

File: PepperiZebraPrinter/app/src/main/java/com/pepperi/printer/view/fragments/Main/MainFragment.kt [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+            adapter = listAdapter
+        }
+    }
+
+    private fun initObservers() {
+        mainViewModel.allPrintersLiveData.observe(viewLifecycleOwner) { list ->
+            Log.e("allPrintersLD.observe", list.toString())
+
+            listAdapter.submitList(list)
+            checkScreenEmptyList()
```
Comment (Chasky-H) [inline PepperiZebraPrinter/app/src/main/java/com/pepperi/printer/view/fragments/Main/MainFragment.kt:73]: put in function refreshList()

File: PepperiZebraPrinter/app/src/main/java/com/pepperi/printer/view/fragments/Main/MainFragment.kt (hunk 1 above)
Comment (Chasky-H) [reply PepperiZebraPrinter/app/src/main/java/com/pepperi/printer/view/fragments/Main/MainFragment.kt:84]: call refreshList do not wait for event from VM

File: PepperiZebraPrinter/app/src/main/java/com/pepperi/printer/model/Repository.kt (hunk 2 above)
Comment (nadavPepperi) [reply PepperiZebraPrinter/app/src/main/java/com/pepperi/printer/model/Repository.kt:20]: build with the sdk

File: PepperiZebraPrinter/app/src/main/java/com/pepperi/printer/view/adapters/ListPrinterAdapter.kt (hunk 3 above)
Comment (nadavPepperi) [reply PepperiZebraPrinter/app/src/main/java/com/pepperi/printer/view/adapters/ListPrinterAdapter.kt:15]: list needed to bind the view

File: PepperiZebraPrinter/app/src/main/java/com/pepperi/printer/view/adapters/ListPrinterAdapter.kt (hunk 4 above)
Comment (nadavPepperi) [reply PepperiZebraPrinter/app/src/main/java/com/pepperi/printer/view/adapters/ListPrinterAdapter.kt:37]: needed to the internal list
